modules/mediacodec_dec/mediacodec_dec.c

```diff
@@ -0,0 +1,891 @@
+#include <gpac/modules/codec.h>
+#include <gpac/internal/media_dev.h>
+#include <gpac/constants.h>
+
+#include "../../src/compositor/gl_inc.h"
+
+#ifdef GPAC_ANDROID
+
+#include <jni.h>
+
+#include "media/NdkMediaCodec.h"
+#include "media/NdkMediaExtractor.h"
+#include "media/NdkMediaFormat.h"
+
+
+#include <android/log.h>
+
+#define TAG "mediacodec_dec"
+
+#define LOGV(...)  __android_log_print(ANDROID_LOG_VERBOSE, TAG,  __VA_ARGS__)
+#define LOGD(...)  __android_log_print(ANDROID_LOG_DEBUG, TAG,  __VA_ARGS__)
+#define LOGE(...)  __android_log_print(ANDROID_LOG_ERROR, TAG,  __VA_ARGS__)
+#define LOGW(...)  __android_log_print(ANDROID_LOG_WARN, TAG,  __VA_ARGS__)
+#define LOGI(...)  __android_log_print(ANDROID_LOG_INFO, TAG,  __VA_ARGS__)
+
+#define DEQUEUE_TIMEOUT 100000
+
+#endif
+
+typedef struct 
+{
+
+#ifdef GPAC_ANDROID
+
+    AMediaCodec *codec;
+    AMediaFormat *format;
+
+#endif
+
+    u32 width, height, stride, out_size;
+    u32 pixel_ar, pix_fmt;
+
+    const char *mime;
+
+    u8 chroma_format, luma_bit_depth, chroma_bit_depth;
+    GF_ESD *esd;
+
+    u32 counter;
+    u16 frame_rate;
+
+    Bool inputEOS, outputEOS;
+        
+    //NAL-based specific
+    char *sps, *pps;
+    u32 sps_size, pps_size;
+    u32 nalu_size_length;
+
+    //hevc
+    char *vps;
+    u32 nb_layers, luma_bpp, chroma_bpp, dec_frames;
+    u8 chroma_format_idc;
+    u16 ES_ID;
+    u32 vps_size;
+
+} MCDec;
+
+
+#ifdef GPAC_ANDROID
+u8 sdkInt() 
+{
+    char sdk_str[3] = "0";
+    __system_property_get("ro.build.version.sdk", sdk_str, "0");
+    return atoi(sdk_str);
+}
+#endif
+
+
+u32 androidPixelFormat(u32 pix_fmt) {
```

Comment:
I think we found the mapping on these values in the Android doc, maybe we should add a comment to precise that.